src/ui/pages/Menu/Menu.tsx

```diff
@@ -17,34 +17,78 @@ import {
   fingerPrintOutline,
   idCardOutline,
 } from "ionicons/icons";
+import { useMemo, useState } from "react";
 import { TabLayout } from "../../components/layout/TabLayout";
 import { useAppDispatch } from "../../../store/hooks";
-import {
-  setCurrentOperation,
-  setCurrentRoute,
-} from "../../../store/reducers/stateCache";
+import { setCurrentRoute } from "../../../store/reducers/stateCache";
 import { TabsRoutePath } from "../../../routes/paths";
 import "./Menu.scss";
 import { i18n } from "../../../i18n";
-import { OperationType } from "../../globals/types";
+import { Settings } from "./components/Settings";
+import { SubMenu } from "./components/SubMenu";
+import { MenuItemProps, SubMenuData, SubMenuKey } from "./Menu.types";
+
+const emptySubMenu = {
+  Component: () => <></>,
+  title: "",
+  additionalButtons: <></>,
+};
+
+const submenuMap = new Map<SubMenuKey, SubMenuData>([
+  [
+    SubMenuKey.Settings,
+    {
+      Component: Settings,
+      title: "settings.sections.header",
+      additionalButtons: <></>,
+    },
+  ],
+]);
+
+const MenuItem = ({ itemKey, icon, label, onClick }: MenuItemProps) => {
+  return (
+    <IonCol>
+      <IonCard
+        onClick={() => onClick(itemKey)}
+        data-testid={`menu-input-item-${itemKey}`}
+        className="menu-input"
+      >
+        <IonIcon
+          icon={icon}
+          color="primary"
+        />
+        <IonLabel>{label}</IonLabel>
+      </IonCard>
+    </IonCol>
+  );
+};
 
 const Menu = () => {
   const pageId = "menu-tab";
   const dispatch = useAppDispatch();
+  const [showSubMenu, setShowSubMenu] = useState(false);
+  const [selectedOption, setSelectedOption] = useState<
+    SubMenuKey | undefined
+  >();
 
   useIonViewWillEnter(() => {
     dispatch(setCurrentRoute({ path: TabsRoutePath.MENU }));
   });
 
+  const showSelectedOption = (key: SubMenuKey) => {
+    if (!submenuMap.has(key)) return;
+
+    setShowSubMenu(true);
+    setSelectedOption(key);
+  };
```

Comment:
I think this could just be
```
  const showSettings = () => {
    setShowSubMenu(true);
    setSelectedOption(SubMenuKey.Settings);
  };
```
as this is all it's used for.

Reply:
`MenuItem` has `onClick` which in all cases is `setSelectedOption` - when this is updated, the `selectSubmenu` useMemo is triggered to update the state.

This function `showSelectedOption` (which I suggested to be `showSettings`) is only used in the additional buttons (the settings icon).

`setSelectedOption` vs `showSelectedOption` :)

Reply:
resolving as `showSelectedOption` meant to be used in both places and is updated